src/Traversals/depthfirst.jl

```diff
@@ -151,3 +151,106 @@ end
 end
 
 @traitfn is_cyclic(g::::(!IsDirected), alg::TraversalAlgorithm=DepthFirst()) = ne(g) > 0
+
+mutable struct BiconnectionState{E <: AbstractEdge} <: TraversalState
+
+    low::Vector{Int}
+    depth::Vector{Int}
+    stack::Vector{E}
+    components::Vector{Vector{E}}
+    id::Int
+    up::Bool
+    lastnode::Int
+    lastlow::Int
+    parent::Vector{Int}
+end
+
+
+@inline function previsitfn!(s::BiconnectionState{E}, u) where E
+    if !s.up 
+        s.up = true
+        s.id+=1
+        s.depth[u] = s.id
+        push!(s.low,s.id)
+    elseif s.depth[u] <= s.lastlow
+        w = s.lastnode
+        e = E(0, 0)  #Invalid Edge, used for comparison only
+        st = Vector{E}()
+        while e != E(min(u, w), max(u, w)) && !isempty(s.stack)
+            e = pop!(s.stack)
+            push!(st, e)
+        end
+        push!(s.components , st)
+    elseif s.low[end] > s.lastlow
+        s.low[end] = s.lastlow
+    end
+    return true
+end
+
+@inline function visitfn!(s::BiconnectionState{E}, u, v) where E
+    (v == s.parent[end] || s.depth[u] < s.depth[v]) && return true
+
+    push!(s.stack, E(min(v, u), max(v, u)))
+    if s.depth[v] != 0
+        s.low[end] = min(s.low[end], s.depth[v])
+    else
+        push!(s.parent,u)
+        s.up = false
+    end
+    return true
+end
+
+@inline function postvisitfn!(s::BiconnectionState{E}, u) where E
+    
+    if s.up
+        s.lastnode = u
+        s.lastlow = pop!(s.low)
+        pop!(s.parent)
+    end
+
+    return true
+end
+
+function BiconnectionState(g)
+    n = nv(g)
+    E = Edge{eltype(g)}
+    return BiconnectionState(Vector{Int}(), zeros(Int, n), Vector{E}(), Vector{Vector{E}}(), 0, false, -1, -1, Vector{Int}())
+end
+
+
+"""
+    biconnected_components(g) -> Vector{Vector{Edge{eltype(g)}}}
+
+Compute the [biconnected components](https://en.wikipedia.org/wiki/Biconnected_component)
+of an undirected graph `g` and return a vector of vectors containing each
+biconnected component.
+### Performance
```

Comment:
blank line between the docstring and the `### Performance` header. Please refer to other docstrings for consistent format.